examples/vanilla-es6/src/store.js

```diff
@@ -52,93 +65,90 @@ export default class Store {
 	}
 
 	/**
-	 * Will save the given data to the DB. If no item exists it will create a new
-	 * item, otherwise it'll simply update an existing item's properties
+	 * Insert an item into the Store.
 	 *
-	 * @param {object} updateData The data to save back into the DB
-	 * @param {function} callback The callback to fire after saving
-	 * @param {number} id An optional param to enter an ID of an item to update
+	 * @param {Item} item Item to insert
+	 * @param {function()} [callback] Called when item is inserted
 	 */
-	save(updateData, callback, id){
-		const data = JSON.parse(localStorage[this._dbName]);
-		const todos = data.todos;
-		const len = todos.length;
-
-		// If an ID was actually given, find the item and update each property
-		if (id) {
-			for (let i = 0; i < len; i++) {
-				if (todos[i].id === id) {
-					for (let key in updateData) {
-						todos[i][key] = updateData[key];
-					}
-					break;
-				}
-			}
+	insert(item, callback) {
+		const todos = this.getLocalStorage();
+		todos.push(item);
+		this.setLocalStorage(todos);
 
-			localStorage[this._dbName] = JSON.stringify(data);
-
-			if (callback) {
-				callback.call(this, JSON.parse(localStorage[this._dbName]).todos);
-			}
-		} else {
-			// Generate an ID
-			updateData.id = new Date().getTime();
-
-			todos.push(updateData);
-			localStorage[this._dbName] = JSON.stringify(data);
-
-			if (callback) {
-				callback.call(this, [updateData]);
-			}
+		if (callback) {
+			callback();
 		}
 	}
 
 	/**
-	 * Will remove an item from the Store based on its ID
+	 * Remove items from the Store based on a query.
 	 *
-	 * @param {number} id The ID of the item you want to remove
-	 * @param {function} callback The callback to fire after saving
+	 * @param {ItemQuery} query Query matching the items to remove
+	 * @param {function(ItemList)|function()} [callback] Called when records matching query are removed
 	 */
-	remove(id, callback){
-		const data = JSON.parse(localStorage[this._dbName]);
-		const todos = data.todos;
-		const len = todos.length;
-
-		for (let i = 0; i < todos.length; i++) {
-			if (todos[i].id == id) {
-				todos.splice(i, 1);
-				break;
+	remove(query, callback) {
+		let k;
+
+		const todos = this.getLocalStorage().filter(todo => {
+			for (k in query) {
+				if (query[k] !== todo[k]) {
+					return true;
+				}
 			}
-		}
+			return false;
+		});
 
-		localStorage[this._dbName] = JSON.stringify(data);
+		this.setLocalStorage(todos);
 
 		if (callback) {
-			callback.call(this, JSON.parse(localStorage[this._dbName]).todos);
+			callback(todos);
 		}
 	}
 
 	/**
-	 * Will drop all storage and start fresh
+	 * Count total, active, and completed todos.
 	 *
-	 * @param {function} callback The callback to fire after dropping the data
+	 * @param {function(number, number, number)} callback Called when the count is completed
 	 */
-	drop(callback){
-		localStorage[this._dbName] = JSON.stringify({todos: []});
+	count(callback) {
+		this.find(emptyItemQuery, data => {
```

Comment:
@xorgy here why not directly get the todos by using `this.getLocalStorage()` here

Reply:
Because once you compile it, the resulting code is smaller, I believe.